config/config.go

```diff
@@ -723,6 +723,29 @@ func (c *RelabelConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 // Regexp encapsulates a regexp.Regexp and makes it YAML marshallable.
 type Regexp struct {
 	regexp.Regexp
+	original string
+}
+
+// NewRegexp creates a new anchored Regexp and returns an error if the
+// passed-in regular expression does not compile.
+func NewRegexp(s string) (*Regexp, error) {
```

Comment:
In other places we usually just make such constructors local the tests if they are necessary to setup table tests.